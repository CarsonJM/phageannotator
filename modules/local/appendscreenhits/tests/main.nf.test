nextflow_process {

    name "Test process: APPENDSCREENHITS"
    script "modules/local/appendscreenhits/main.nf"
    process "APPENDSCREENHITS"
    tag "modules"
    tag "modules_local"
    tag "appendscreenhits"


    test("Should run without failures") {

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id: 'test' ], [ file(params.test_data['modules_local']['mash_screen_results_tsv'], checkIfExists: true) ],
                            [ file(params.test_data['modules_local']['assembly_fasta_gz'], checkIfExists: true) ] ]
                input[1] = [ [ id: 'reference' ], [ file(params.test_data['modules_local']['reference_fasta_gz'], checkIfExists: true) ] ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
