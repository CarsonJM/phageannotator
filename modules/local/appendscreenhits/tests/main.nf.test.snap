{
    "['modules_local']['mash_screen_results_tsv'] & ['modules_local']['assembly_fasta_gz'] & ['modules_local']['reference_fasta_gz'] - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta_w_screen_hits.fna.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,65b56210fc3597cf5d869e5a1384a1a3"
                ],
                "assembly_w_screen_hits": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta_w_screen_hits.fna.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,65b56210fc3597cf5d869e5a1384a1a3"
                ]
            }
        ],
        "timestamp": "2024-01-17T22:50:26.825206996"
    },
    "['modules_local']['mash_screen_results_tsv'] & ['modules_local']['assembly_fasta_gz'] & ['modules_local']['reference_fasta_gz']": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta_w_screen_hits.fna.gz:md5,8ae1a1f526134b2672776111c9f9f210"
                    ]
                ],
                "1": [
                    "versions.yml:md5,65b56210fc3597cf5d869e5a1384a1a3"
                ],
                "assembly_w_screen_hits": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fasta_w_screen_hits.fna.gz:md5,8ae1a1f526134b2672776111c9f9f210"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,65b56210fc3597cf5d869e5a1384a1a3"
                ]
            }
        ],
        "timestamp": "2023-12-06T15:23:52.515798496"
    }
}